lib/Model/Settings/AppSettings.php

```diff
@@ -132,6 +132,65 @@ public function getComboAllowed(): bool {
 		return false;
 	}
 
+	/**
+	 * Permission to create shares is controlled by core settings
+	 */
+	public function getShareCreateAllowed(): bool {
+		if (!$this->userSession->getIsLoggedIn()) {
+			// only logged in users can create shares
+			return false;
+		}
+
+		// first check group exception mode
+		$groupExceptionMode = $this->getShareGroupExceptionMode();
+
+		if ($groupExceptionMode === 'off') {
+			// no group exceptions
+			return true;
+		}
+
+		// get group exceptions
+		$excludedGroups = $this->getShareExceptionGroups();
+
+		if ($groupExceptionMode === 'allowGroup') {
+			// exception mode is 'limit sharing to some groups'
+			// if user is in exception group, allow share creation
+			return $this->userSession->getUser()->getIsInGroupArray($excludedGroups);
+		}
+
+		// exception mode is 'Exclude some Groups from sharing'
+		// if user is in exception group, deny share creation
+		return $this->userSession->getUser()->getIsInGroupArray($excludedGroups);
+
+	}
+
+	/**
+	 * Get share exception groups
+	 * @return array
+	 */
+	public function getShareExceptionGroups(): array {
+		$excludedGroups = $this->config->getAppValue('core', 'shareapi_exclude_groups_list', '');
+		return json_decode($excludedGroups, true) ?? [];
+	}
+
+	/**
+	 * Get share group exception mode
+	 * @return string
+	 * @psalm-return 'denyGroup'|'allowGroup'|'off'
+	 * Take value from the core setting 'shareapi_exclude_Mode' and translate
+	 * 'yes' => 'denyGroup' ('Exclude some groups from sharing')
+	 * 'allow' => 'allowGroup' (Limit sharing to some groups)
+	 * default => 'off' (Allow sharing for everyone) or setting absent
+	 */
+	public function getShareGroupExceptionMode(): string {
+		$excludedMode = $this->config->getAppValue('core', 'shareapi_exclude_Mode', '');
```

Comment:
```suggestion
		$excludedMode = $this->config->getAppValue('core', 'shareapi_exclude_groups', '');
```

Reply:
Hey @hamza221 Thanks for reviewing. But you may save time if you wait until I am finished with the next commit. If my discussion is solved, I will implement the final logic which will be ready for review and testing. 

Just want to avoid duplicated work.

Reply:
Already fixed and enhanced.